### PR TITLE
ci: update vscode-test and add logging for tests

### PR DESCRIPTION
**Pull request by gbockus-sf**
### What does this PR do?
Update the vscode-test dependancy.
Alter a few lwc vscode-integration tests to make them more reliable. 
Add Logging to the integration test script so we can track where they fail when they do. 

### What issues does this PR fix or reference?
@W-12216803@

### Functionality Before
Every GHA int test run fails.

### Functionality After
GHA int test runs succeed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "@salesforce/dev-config": "^3.1.0",
     "@tsconfig/node14": "1.0.3",
     "@types/jest": "28.1.7",
+    "@vscode/test-electron": "2.2.1",
```
Comment (gbockus-sf) [inline package.json:24]: vscode-test was migrated to this npm module.

File: packages/salesforcedx-vscode-lwc/src/constants.ts [hunk 2]
```diff
@@ -7,3 +7,7 @@
 
 export const ESLINT_NODEPATH_CONFIG = 'eslint.nodePath';
 export const LWC_EXTENSION_NAME = 'salesforcedx-vscode-lwc';
+
+export const log = (message: string) => {
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-lwc/src/constants.ts:11]: the test was stubbing the console.log method and it never seemed to work correctly.  I moved it out to this wrapper method so the log call could be stubbed.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts [hunk 3]
```diff
@@ -10,12 +11,11 @@ import { expect, assert } from 'chai';
...
 
 describe('activation modes', () => {
-  let sandbox: sinon.SinonSandbox;
+  const sandbox = sinon.createSandbox();
   let mockExtensionContext: vscode.ExtensionContext;
 
   beforeEach(function() {
-    sandbox = sinon.createSandbox();
-    sandbox.spy(console, 'log');
+    sandbox.spy(constants, 'log');
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts:18]: update to use the log wrapper.

File: scripts/install-vsix-dependencies.js [hunk 4]
```diff
@@ -7,16 +7,23 @@ const shell = require('shelljs');
...
+);
+
+console.log('### where is VScode', { vscodeDownloadDir });
 // VSCode no longer downloads to a single directory name like 'stable'. The folder
 // name is dynamic base on the version number, so lets just use the first folder in .vscode-test dir
 // as the assumed place where vscode is extracted
 const testRunFolder = path.join(
   '.vscode-test',
-  isInsiders ? 'insiders' : fs.readdirSync(`${process.cwd()}/.vscode-test`)[0]
+  isInsiders ? 'insiders' : vscodeDownloadDir
```
Comment (gbockus-sf) [inline scripts/install-vsix-dependencies.js:26]: I'll remove the above comment.  This wasn't true for multiple runs of the int tests and ofter resulted in failure locally.

File: scripts/download-vscode.js [hunk 5]
```diff
@@ -79,7 +91,7 @@ downloadAndUnzipVSCode(vscodeVersion)
     }
   })
   .catch(err => {
-    logger.err('Failed to download vscode');
+    logger.error('Failed to download vscode');
```
Comment (gbockus-sf) [inline scripts/download-vscode.js:94]: this method didn't exist.

File: scripts/download-vscode.js [hunk 6]
```diff
@@ -50,8 +55,15 @@ downloadAndUnzipVSCode(vscodeVersion)
     // Example:
     // ~/salesforcedx-vscode/packages/salesforcedx-vscode-lwc> npm run test:vscode-integration
     if (vscodeBasePath.indexOf(path.sep + 'packages' + path.sep) !== -1) {
+      console.log('### vscode package install found');
       // Do nothing, vscode is already downloaded and extracted in this package
+
+      const userData = path.join(vscodeBasePath, VSCODE_TEST, USER_DATA);
+      console.log('### userdata: ' + userData);
+      // Remove the previously generated user data to ensuer we can run the int tests
+      fs.rmSync(userData, { recursive: true, force: true });
```
Comment (gbockus-sf) [inline scripts/download-vscode.js:64]: This is probably fallout from the upgrade.  basically there a user-data directory that gets created in the .vscode-test dir. It points a directory that causes issues on re-runs.  This fixes that problem.  I'll fix the ensuer typo in the comment above.

File: scripts/download-vscode.js [hunk 7]
```diff
@@ -50,8 +55,15 @@ downloadAndUnzipVSCode(vscodeVersion)
...
     if (vscodeBasePath.indexOf(path.sep + 'packages' + path.sep) !== -1) {
+      console.log('### vscode package install found');
       // Do nothing, vscode is already downloaded and extracted in this package
+
+      const userData = path.join(vscodeBasePath, VSCODE_TEST, USER_DATA);
+      console.log('### userdata: ' + userData);
+      // Remove the previously generated user data to ensure we can run the int tests
+      fs.rmSync(userData, { recursive: true, force: true });
     } else {
+      console.log('### in the else');
```
Comment (klewis-sfdc) [inline scripts/download-vscode.js:66]: Do you want to keep all of these console.log() statements?  Most look helpful.  Wondering about this one and maybe the userdata one above.

**Review by klewis-sfdc — APPROVED** [review]
Looks great, Gordon.  Just one Q from me about the console.log statements.

File: scripts/download-vscode.js (hunk 7 above)
Comment (gbockus-sf) [reply scripts/download-vscode.js:66]: Good call.  I'll clean up those two and leave the rest.  They are helpful for seeing the flow of the int test run

**Pull request merged**